lib/kcerts/ssh.go

```diff
@@ -191,8 +241,33 @@ func FindSSHAgent(store cache.Store, logger logger.Logger) (*SSHAgent, error) {
 	if err != nil {
 		return nil, err
 	}
-	logger.Infof("%s", WriteAgentToCache(store, newAgent))
-	return newAgent, nil
+	if newAgent != nil && newAgent.Valid() {
+		return newAgent, nil
+	}
+	return nil, fmt.Errorf("Could not find or start an ssh-agent")
+}
+
+// PrepareSSHAgent ensures that we end up with a working ssh-agent,
+// either by discovering an existing ssh-agent or creating a new one.
+// It also ensures that we have an up-to-date symlink to that agent's
+// socket in the standard location.
+//
+// The final ssh-agent socket returned by PrepareSSHAgent is always
+// ~/.config/enkit/agent.
+func PrepareSSHAgent(store cache.Store, logger logger.Logger) (*SSHAgent, error) {
+	agent, err := FindOrCreateSSHAgent(store, logger)
+	if err != nil {
+		return nil, err
```

Comment:
Many unannotated errors returned through this code. Adding on some form of context can provide a sort of "poor man's backtrace" and make it easier to understand at a glance what happened/find the precise codepath that resulted in the error:

```
return nil, fmt.Errorf("while attaching to SSH agent: %w", err)
```

can result in:

```
while attaching to SSH agent: while creating new SSH agent: unable to delete existing socket: foo
```

instead of just the last part, passed all the way up